src/frontend/src/lib/utils/format.utils.ts

```diff
@@ -40,6 +40,18 @@ export const formatTokenDetailed = ({
 		.replace(/(\.0+|0+)$/, '');
 };
 
+export const formatNumberAsCurrency = ({
+	value,
+	displayDecimals = 2
+}: {
+	value: number;
+	displayDecimals?: number;
+}): string => {
+	return new Intl.NumberFormat('en-US', { minimumFractionDigits: displayDecimals })
+		.format(value)
+		.replace(/(\.0+|0+)$/, '');
```

Comment:
sounds like the same pattern has above, can we refactor instead of duplicating the code?